velox/core/PlanNode.cpp

```diff
@@ -152,6 +190,96 @@ void AggregationNode::addDetails(std::stringstream& stream) const {
   }
 }
 
+// static
+const char* AggregationNode::stepName(AggregationNode::Step step) {
+  switch (step) {
+    case Step::kPartial:
+      return "PARTIAL";
+    case Step::kFinal:
+      return "FINAL";
+    case Step::kIntermediate:
+      return "INTERMEDIATE";
+    case Step::kSingle:
+      return "SINGLE";
+    default:
+      VELOX_UNREACHABLE();
+  }
+}
+
+// static
+AggregationNode::Step AggregationNode::stepFromName(const std::string& name) {
+  static const std::unordered_map<std::string, AggregationNode::Step> kSteps = {
+      {"PARTIAL", Step::kPartial},
+      {"FINAL", Step::kFinal},
+      {"INTERMEDIATE", Step::kIntermediate},
+      {"SINGLE", Step::kSingle}};
```

Comment:
nit: putting a comma after the last element usually makes clang-format it in a nicer way.